PropertyManagerMyApp/PropertyManagerMyApp/Controllers/O365CalendarController.cs

```diff
@@ -17,126 +18,76 @@ public class O365CalendarController : Controller
-
-                    if (beginTimeSpan >= start && endTimeSpan <= end)
-                    {
-                        timeSlots.RemoveAt(i);
-                    }
-                }
-            }
+                Start = availableStart.Hours > 9 ? availableStart.Hours.ToString() : string.Format("0{0}", availableStart.Hours),
+                Value = string.Format("{0} - {1}", availableStart.ToString(@"hh\:mm"), availableEnd.ToString(@"hh\:mm"))
+            };
+            timeSlots.Add(timeSlot);
         }
 
-        private async Task<List<CalendarEvent>> GetEventsByRestAPI()
+        private async Task<List<Event>> GetEventsByRestAPI()
         {
-            var result = new List<CalendarEvent>();
-            var accessToken = AuthenticationHelper.GetGraphAccessTokenAsync();
-            var restURL = string.Format("{0}/me/events", AADAppSettings.GraphResourceUrl);
-            try
-            {
-                using (HttpClient client = new HttpClient())
-                {
-                    var accept = "application/json";
-
-                    client.DefaultRequestHeaders.Add("Accept", accept);
-                    client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", await accessToken);
-
-                    using (var response = await client.GetAsync(restURL))
-                    {
-                        if (response.IsSuccessStatusCode)
-                        {
-                            var jsonresult = JObject.Parse(await response.Content.ReadAsStringAsync());
-
-                            foreach (var item in jsonresult["value"])
-                            {
-                                result.Add(new CalendarEvent
-                                {
-                                    Start = !string.IsNullOrEmpty(item["start"]["dateTime"].ToString()) ? DateTime.Parse(item["start"]["dateTime"].ToString()) : new DateTime(),
-                                    End = !string.IsNullOrEmpty(item["end"]["dateTime"].ToString()) ? DateTime.Parse(item["end"]["dateTime"].ToString()) : new DateTime(),
-                                    ShowAs = !string.IsNullOrEmpty(item["showAs"].ToString()) ? item["showAs"].ToString() : string.Empty
-                                });
-                            }
-                        }
-
-                        if (response.StatusCode == System.Net.HttpStatusCode.Unauthorized)
-                            throw new System.Web.Http.HttpResponseException(System.Net.HttpStatusCode.Unauthorized);
-                    }
-                }
-            }
-            catch (Exception el)
-            {
-                el.ToString();
-            }
-
-            return result;
+            GraphServiceClient graphServiceClient = await AuthenticationHelper.GetGraphServiceAsync(AADAppSettings.GraphResourceUrl);
+            var queryEvents = await graphServiceClient.Me.Events.Request().GetAllAsnyc();
```

Comment:
Typo here - this would not compile.   I don't think this codepath from the Index() is exercised.